src/views/partials/breach-detail.js

```diff
@@ -0,0 +1,206 @@
+    const dataType = getMessage(dataClass)
+    if (priorityDataClasses[dataClass]) {
+      priorityDataClasses[dataClass].dataType = dataType
+      sortedDataClasses.priority.push(priorityDataClasses[dataClass])
+      return
+    }
+    sortedDataClasses.lowerPriority.push(dataType)
+  })
+  sortedDataClasses.priority.sort((a, b) => { return b.weight - a.weight })
+  sortedDataClasses.lowerPriority = sortedDataClasses.lowerPriority.join(', ')
+  return sortedDataClasses
+}
+
+function makeDataSection (breach) {
+  const dataClasses = getSortedDataClasses(breach)
+
+  let output = dataClasses.priority.map(dataClass =>
+    `<li>
+      <img src="${dataClass.pathToGlyph}.svg" width="24" alt="">
+      ${dataClass.dataType}
+    </li>`
+  ).join('')
+
+  output +=
+    `<li>
+      <img src="/images/more.svg">
+      ${dataClasses.lowerPriority}
+    </li>`
+
+  return output
+}
+
+function makeRecommendationCards (breach) {
+  const dataClasses = getSortedDataClasses(breach)
+
+  let output = dataClasses.priority.map(dataClass =>
+    dataClass.recommendations?.map(r =>
+    `<div class="breach-detail-recommendation ${r.recIconClassName}">
+      <dt>${getMessage(r.recommendationCopy.subhead)}</dt>
+      <dd>
+        <p>${getMessage(r.recommendationCopy.body)}</p>
+        ${r.recommendationCopy.cta ? `<a href="${r.ctaHref}" target="${r.ctaShouldOpenInNewTab ? '_blank' : '_self'}">${getMessage(r.recommendationCopy.cta)}</a>` : ''}
+      </dd>
+    </div>`).join('')
+  ).join('')
+
+  output += getAllGenericRecommendations().map(dataClass =>
+    `<div class="breach-detail-recommendation ${dataClass.recIconClassName}">
+      <dt>${getMessage(dataClass.recommendationCopy.subhead)}</dt>
+      <dd>
+        <p>${getMessage(dataClass.recommendationCopy.body)}</p>
+        ${dataClass.recommendationCopy.cta ? `<a href="${dataClass.ctaHref}" target="${dataClass.ctaShouldOpenInNewTab ? '_blank' : '_self'}">${getMessage(dataClass.recommendationCopy.cta)}</a>` : ''}
+      </dd>
+    </div>`
+  ).join('')
+
+  return output
+}
+
+function getBreachDetail (categoryId) {
+  if (categoryId === 'data-aggregator-breach') {
+    return {
+      subhead: getMessage('what-is-data-agg'),
+      body: getMessage('what-is-data-agg-blurb')
+
+    }
+  } else if (categoryId === 'sensitive-breach') {
+    return {
+      subhead: getMessage('sensitive-sites'),
+      body: getMessage('sensitive-sites-copy')
+
+    }
+  } else {
+    return {
+      subhead: getMessage('what-is-a-website-breach'),
+      body: getMessage('website-breach-blurb')
+    }
+  }
+}
+
+function makeBreachDetails (breach) {
+  const breachDetail = getBreachDetail(breach)
+  return `
+  <h2>${breachDetail.subhead}</h2>
+  <div>${breachDetail.body}</div>
+  `
+}
+
+export const breachDetailsPartial = data => `
+  <header class="breach-detail-header">
+    <img class="breach-detail-logo breach-logo" alt="" src="https://monitor.cdn.mozilla.net/img/logos/${data.breach.LogoPath}" />
```

Comment:
Q: can you confirm if this is correct. Are we still serving breach logos from the CDN or were you fetching and caching those locally now?

Reply:
Yes we're still using the CDN for this, these are larger company logos whereas the ones we fetch and cache locally are [favicons](https://en.wikipedia.org/wiki/Favicon).

We should definitely pick a lane here, I think the ideal would be if we start using a CDN for everything :) I'll file a ticket about this.

Reply:
Filed https://mozilla-hub.atlassian.net/browse/MNTOR-1465

Reply:
Well, now I'm right confused.
I thought the current (v1) process is that @maxxcrawford lovingly creates a bespoke, artisanal 150&times;150 logo and submits a PR and somebody approves it and it gets hoisted into the CDN.
My understanding is that the new (v2) process did away with the CDN-ed, free-range images and only used the faviconed images (thus getting rid of the need for Maxx to make custom high quality logos).

I'm down with somebody finding a way to move the faviconed images into a single CDN solution (vs each machine in a cluster downloading and managing their own logos), but just wanted to confirm whether or not Maxx is still expected to toil away in the logo mines each time we need a logo.

Reply:
Ah yeah I assumed we were falling forward into the favicon automated route – I didn't have a chance to review these designs beforehand. I think we can release this now for the SEO, knowing Favicons on CDN is the way for future.